libfoundation/src/foundation-string.cpp

```diff
@@ -1766,8 +1854,11 @@ bool MCStringUnmapGraphemeIndices(MCStringRef self, MCLocaleRef p_locale, MCRang
 
 bool MCStringUnmapTrueWordIndices(MCStringRef self, MCLocaleRef p_locale, MCRange p_in_range, MCRange &r_out_range)
 {
-    MCAssert(self != nil);
-    MCAssert(p_locale != nil);
+	__MCAssertIsString(self);
+	__MCAssertIsLocale(p_locale);
+
+	if (__MCStringIsIndirect(self))
+		self = self->string;
     
```

Comment:
You can make the subsequent `MCStringGetLength`s `__MCStringGetLength`s now, as it's no longer indirect.